src/System.Console/src/System/TermInfo.cs

```diff
@@ -346,21 +349,22 @@ public int GetNumber(WellKnownNumbers numberIndex)
             /// defined as the earlier portions, and may not even exist, the parsing is more lenient about
             /// errors, returning an empty collection rather than throwing.
             /// </returns>
-            private static Dictionary<string, string> ParseExtendedStrings(byte[] data, int extendedBeginning)
+            private static Dictionary<string, string> ParseExtendedStrings(byte[] data, int extendedBeginning, bool readAs32Bit)
             {
                 const int ExtendedHeaderSize = 10;
+                int sizeOfIntValuesInBytes = (readAs32Bit) ? 4 : 2;
                 if (extendedBeginning + ExtendedHeaderSize >= data.Length)
                 {
                     // Exit out as there's no extended information.
                     return null;
                 }
 
                 // Read in extended counts, and exit out if we got any incorrect info
-                int extendedBoolCount = ReadInt16(data, extendedBeginning);
-                int extendedNumberCount = ReadInt16(data, extendedBeginning + 2);
-                int extendedStringCount = ReadInt16(data, extendedBeginning + 4);
-                int extendedStringNumOffsets = ReadInt16(data, extendedBeginning + 6);
-                int extendedStringTableByteSize = ReadInt16(data, extendedBeginning + 8);
+                int extendedBoolCount = ReadInt(data, extendedBeginning, readAs32Bit);
+                int extendedNumberCount = ReadInt(data, extendedBeginning + (sizeOfIntValuesInBytes * 1), readAs32Bit);
+                int extendedStringCount = ReadInt(data, extendedBeginning + (sizeOfIntValuesInBytes * 2), readAs32Bit);
+                int extendedStringNumOffsets = ReadInt(data, extendedBeginning + (sizeOfIntValuesInBytes * 3), readAs32Bit);
+                int extendedStringTableByteSize = ReadInt(data, extendedBeginning + (sizeOfIntValuesInBytes * 4), readAs32Bit);
```

Comment:
Are you sure all of these change? It seems like the difference is only about the size of the numbers in the numbers section, not all 16-bit values in the database.